runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -59,6 +59,13 @@
 
   private Environments() {}
 
+  public static Environment createEnvironment(String url){
```

Comment:
It's weird to require a string that will be ignored in most cases. I would either add an override or a separate method that just returns the default. Since we're already reading environment variables for the default image url, it may be reasonable to pull the override url from an environment variable as well.

Reply:
Not exactly, the environment URL is used for fusion and is key component in deciding which container to run a given operation.
This information is mute in the pre portability runners but should be used appropriately in portability runners.